Bootloader/Bootloader/Mcal/Adc/Adc.c

```diff
@@ -1,6 +1,222 @@
-/*
- * Adc.c
- *
- * Created: 2/7/2020 8:21:58 PM
- *  Author: MahmoudMorsy
- */ 
+/**************************************************************************************************
+* FILE INFORMATION
+* -------------------------------------------------------------------------------------------------
+* NAME:              Adc.c
+* MODULE:            Adc
+* AUTHOR:            ShroukRashwan
+* DESCRIPTION:       Adc driver functional implementation
+* CREATION DATE:     7/2/2020
+* MODIFICATION DATE: 12/2/2020
+**************************************************************************************************/
+
+/**************************************************************************************************
+*                                       INCLUDES & LIBRARIES                                      *
+**************************************************************************************************/
+#include "Adc.h"
+#include "Adc_Internal.h"
+
+/**************************************************************************************************
+*                                         GLOBAL VARIABLES                                        *
+**************************************************************************************************/
+#if UNIT_TESTING == STD_ON
+	uint8 Adc_ADMUX																				=0;
+	uint8 Adc_ADCSRA																			=0;
+	uint8 Adc_ADCH																				=0;
+	uint8 Adc_ADCL																				=0;
+	uint8 Adc_SFIOR																				=0;
+#endif
+
+uint16 Adc_Value 																				=0;
+
+
+/**************************************************************************************************
+*                                     FUNCTIONS IMPLEMENTATION                                    *
+**************************************************************************************************/
+boolean Adc_Init(uint8 Param_ChannelNumber, uint8 Param_VolageReference, uint8 Param_PreScaler)
+{
+	if (Param_ChannelNumber>=0 && Param_ChannelNumber <=7 && Param_VolageReference >=0 &&
```

Comment:
It's better to encapsulate each condition in a separate bracket to avoid MISRA warnings and compiler dependencies